### ISSUE TITLE
DL3009 should ignore tmpfs mounts for apt cache

### ISSUE THREAD

**Issue by detly** [hadolint/hadolint#1095]
I have a Dockerfile where I use a `tmpfs` mount for the apt cache. This is because the image is built in a Docker-in-Docker image in CI, so any cache volumes will be removed when the DinD image goes away at the end of the job anyway.

DL3009 shouldn't trigger for this, since the end result is the same: the apt cache is not in the final image.

Example Dockerfile:

```dockerfile
FROM debian:12

# PASS
# Received: [DL3009]
# Expected: [DL3009]
RUN apt-get update

# PASS
# Received: []
# Expected: []
RUN --mount=type=cache,target=/var/cache/apt --mount=type=cache,target=/var/lib/apt apt-get update

# FAIL
# Received: [DL3009]
# Expected: []
RUN --mount=type=tmpfs,target=/var/cache/apt --mount=type=tmpfs,target=/var/lib/apt apt-get update
```

Output:

```text
-:6 DL3009 info: Delete the apt-get lists after installing something
-:16 DL3009 info: Delete the apt-get lists after installing something
```

Version: Haskell Dockerfile Linter 2.12.1

**Comment by m-ildefons** [comment]
Fixed since v2.14.0
